src/interfaces/mining.h

```diff
@@ -42,8 +42,26 @@ class BlockTemplate
     // Sigop cost per transaction, not including coinbase transaction.
     virtual std::vector<int64_t> getTxSigops() = 0;
 
+    /** Return fields needed to construct a coinbase transaction */
```

Comment:
Given [sv2-tp](https://github.com/stratum-mining/sv2-tp) and a few others are the users of this interface and they are still in development.

I'd rather have any un-useful methods pruned.
When we release a stable interface then we can start deprecating things to avoid breaking clients (As you mentioned in the tracking issue it is okay to make breaking changes so all these methods should be pruned and clients should update).

Reply:
Hmm, do you plan on deprecating it first and then removing it in a later release? On the contrary, I think it should be dropped as soon as we deem it unnecessary. Otherwise, we’ll end up with interface methods that conflict with each other. It would also be easier to review the changes together, rather than keeping them all here with deprecation comments and later deleting them it will save us review cycle.

> Even for testing it's nice that current clients works against both v30 and master.

I think this is not big of deal, since you have a release that is against v30, your master should be  in sync with master which can easily update when breaking change occur.

Reply:
> do you plan on deprecating it first and then removing it in a later release?

No, because the interface is marked experimental I think it's fine to drop it in any release we want. That might be as soon as v31, but we can also do it later.

> your master should be in sync with master

`sv2-tp` is used by people who only run v30, so it has to support both master and v30.

It can already handle _new_ methods by just trying them and falling back to old methods when they're missing. 

Recent versions of `sv2-tp` all work against v30, so I haven't had to push people to switch to a newer version either. But we start making breaking changes, then I have to start instructing users of Bitcoin Core master to use specific newer versions of `sv2-tp`.

Reply:
> In the longer run it probably makes sense for sv2-tp to have one major release of sv2-tp for every major Bitcoin Core release, but current it doesn't have backports. Each release is a combination of bug fixes and new features.


Hmm, fwiw I am approach NACK changes that introduce conflicting interface method that deprecate and don't prune on experimental interface that did not offer interface stability for the reason discussed above.
However I am open minded to convincing otherwise and curious to know what other contributors think.

Reply:
The old methods are redundant, but they're not conflicting. Clients can use either and both have test coverage.

From this projects perspective there's hardly any additional review burden when we prune deprecated methods in a later PR, that should be trivial.